packages/grid/apps/network/src/__main__.py

```diff
@@ -50,16 +50,26 @@
     help="If this flag is used a SQLAlchemy DB URI is generated to use a local db.",
 )
 
+parser.add_argument(
+    "--use-websockets",
+    dest="use_websockets",
+    action="store_true",
+    help="If this flag is used websockets will be used instead of HTTP.",
+)
+
 parser.set_defaults(use_test_config=False)
 
 
 if __name__ == "__main__":
     args = parser.parse_args()
 
-    app = create_app(args)
+    app, socketio = create_app(args)
     _address = "http://{}:{}".format(args.host, args.port)
 
-    server = pywsgi.WSGIServer(
-        (args.host, args.port), app, handler_class=WebSocketHandler
-    )
-    server.serve_forever()
```

Comment:
does this work together in this way? Or is the server.serve_forever() blocking the socketio.run() call?

Reply:
The server setup and serve_forever call are both handled exclusively in the else branch and therefore don't interfere with the socketio.run() in the if(args.use_websockets) branch. My tests also confirm that it's working since the else branch is never called if websockets are used.